src/cache/base.py

```diff
@@ -0,0 +1,102 @@
+from functools import wraps
+from typing import Any, Callable
+from sys import getsizeof
+import pickle
+import time
+
+
+class BaseCache:
+    def __init__(self) -> None:
+        # these specific elements can be change subClasses
+        self.dumper = pickle.dumps
+        self.loader = pickle.loads
+
+    def _log_cache_failed(self, key: str, logger: Any) -> None:
+        """Just some logging"""
+        logger.debug(f">> cache failed on custom_condition for {key=}")
```

Comment:
Curiosity: waht does the `=` means in the formatting?

Reply:
f"{key=}" -> "value"
f"{key=}" -> "key=value"

can be interesting while debugging, it just add the variable name